src/main/java/chatbot/Command.java

```diff
@@ -0,0 +1,171 @@
+package chatbot;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import chatbot.exceptions.DukeException;
+import chatbot.exceptions.InvalidArgumentException;
+
+/**
+ * Represents an enum of the valid user commands.
+ */
+public enum Command {
+    EXIT("bye"),
+    LIST("list"),
+    MARK("mark"),
+    UNMARK("unmark"),
+    DELETE("delete"),
+    TODO("todo"),
+    DEADLINE("deadline"),
+    EVENT("event"),
+    FIND("find");
+
+    private final String rep;
+    private String args = "";
+    private final boolean isBreaking;
+
+    /**
+     * Constructor that initialises a user command.
+     *
+     * @param cmd The string input representing the command.
+     */
+    Command(String cmd) {
+        this.rep = cmd.toLowerCase();
+        this.isBreaking = this.rep.equals("bye");
+    }
+
+    /**
+     * Populates the command with arguments provided.
+     * @param args The arguments.
+     */
+    public void withArgs(String args) {
+        this.args = args;
+    }
+
+    @Override
+    public String toString() {
+        return this.rep + this.args;
+    }
+
+    /**
+     * Getter for if the command breaks the input loop.
+     *
+     * @return True if the command breaks the input loop.
+     */
+    public boolean shouldExit() {
+        return this.isBreaking;
+    }
+
+    /**
+     * Executes the command for the provided task list.
+     *
+     * @param view The UI object to interface with.
+     * @param tl The task list object to operate on.
+     * @throws DukeException If an error occurs with the execution.
+     */
+    public void execute(Ui view, TaskList tl) throws DukeException {
+        switch (this) {
+        case LIST:
+            view.displayList(tl);
+            break;
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            executeAdd(view, tl);
+            break;
+        case MARK:
+        case UNMARK:
+        case DELETE:
+            executeIndexing(view, tl);
+            break;
+        case FIND:
+            executeFind(view, tl);
+            break;
+        default:
+            break;
+        }
+    }
+
+    /**
+     * Executes a command with arguments for the provided task list.
+     *
+     * @param view The UI object to interface with.
+     * @param tl The task list object to operate on.
+     * @throws DukeException If an error occurs with the execution.
+     */
+    private void executeAdd(Ui view, TaskList tl) throws DukeException {
+        Task t = new Task("");
```

Comment:
What is the purpose of this line?

Reply:
To initialise a default task, in case none of the switch cases below are applicable, should not happen in normal situations.